admin-manual/security/csp-headers.rst

```diff
@@ -0,0 +1,263 @@
+===============================
+Content Security Policy Headers
+===============================
+
+Content Security Policy (`CSP`_) headers are a critical component of web security,
+serving as a robust defense against a wide range of online threats. These
+headers, which are implemented on web servers and embedded within web pages,
+define a set of directives that instruct a browser on how to handle and restrict
+the execution of various web resources, such as scripts, stylesheets, and
+images.
+
+The primary function of CSP headers is to mitigate the risks associated with
+cross-site scripting (`XSS`_) attacks, data injection attacks, and other malicious
+activities that exploit the trust model of web applications. By specifying which
+sources of content are permitted, CSP headers help prevent unauthorized code
+execution and reduce the attack surface, enhancing the overall security posture
+of web applications.
+
+Starting in AtoM 2.8, AtoM has built in support for CSP headers when Bootstrap 5
+based themes are in use. When installing AtoM 2.8 from scratch, there's
+a pre-configured CSP directive setting in place and CSP headers will be in use by
+default. These default settings serve as a foundation compatible with the upgraded
+Bootstrap 5 based Dominion theme and can be used as a base for custom themes based
+on Dominion.
+
+Content Security Policy headers
+-------------------------------
+
+In a high-level example of web security using Content Security Policy (CSP),
+let's consider a scenario. When a user's web browser sends a request to a server
+for a particular web page, the server responds by generating the page response
+and including a CSP header. This CSP header specifies trusted sources from which
+scripts, stylesheets, images, and other assets can be loaded and executed.
+Additionally, the server generates a unique nonce (number used once) and adds it
+to all inline scripts and other assets within the response.
+
+Once the response is sent to the user's browser, the browser acts as a security
+enforcer. It checks each requested resource and script against the CSP
+directive. If any resource or script does not match the trusted sources
+specified in the CSP header or lack the correct nonce, the browser rejects those
```

Comment:
nitpick: should it be "lacks" rather than lack here? hard to keep track of...

Reply:
Yes - that sounds better - fixed.